### ISSUE TITLE
Feature: Allow same load balancer for control plane and workloads

### ISSUE THREAD

**Issue by ctholho** [kube-hetzner/terraform-hcloud-kube-hetzner#493]
Would it be possible to add an option that toggles if you would want to use the load balancer also as control plane load balancer? Which would mean that the normal load balancer also gets port 6443 and other related configs from a dedicated control plane load balancer.

Are there good reasons not to allow it? I would try myself on a PR if people are not too concerned about this set up.

**Comment by captnCC** [comment]
An interesting idea to save some cost for HA setups. 

Yet it has a major catch, since the control plane load-balancer is managed by terraform and the ingress load-balancer by the ccm. So we can't just add the port to it for the kube-api. The only solution I see right now, is to publish the kube-api service via ingress. 

If you find a solution to realise it this way, please share! Since this would be only done at Kubernetes level, it should be easily be switched on/off or be provided as an example

**Comment by mysticaltech** [comment]
I do not think this could be done, but I never say never, and PR is welcome as always! :)  For now, moving to discussions.
